packages/babel-plugin-transform-optional-chaining/README.md

```diff
@@ -0,0 +1,114 @@
+# babel-plugin-transform-optional-chaining
+
+The Optional Chaining Operator allows you to handle properties of deeply nested
+objects without worrying about undefined intermediate objects.
+
+## Example
+
+### Accessing deeply nested properties
+
+```js
+const obj = {
+  foo: {
+    bar: {
+      baz: 42,
+    },
+  },
+};
+
+const baz = obj?.foo?.bar?.baz; // 42
+
+const safe = obj?.qux?.baz; // undefined
+
+// Optional chaining and normal chaining can be intermixed
+obj?.foo.bar?.baz; // Only access `foo` if `obj` exists, and `baz` if
+                   // `bar` exists
+```
+
+### Calling deeply nested functions
+
+```js
+const obj = {
+  foo: {
+    bar: {
+      baz() {
+        return 42;
+      },
+    },
+  },
+};
+
+const baz = obj?.foo?.bar?.baz(); // 42
+
+const safe = obj?.qux?.baz(); // undefined
+const safe2 = obj?.foo.bar.qux?.(); // undefined
+
+// Top function can be called directly, too.
+function test() {
+  return 42;
+}
+test?.(); // 42
+
+exists?.(); // undefined
+```
+
+### Constructing deeply nested classes
+
+```js
+const obj = {
+  foo: {
+    bar: {
+      baz: class {
+      },
+    },
+  },
+};
+
+const baz = new obj?.foo?.bar?.baz(); // baz instance
+
+const safe = new obj?.qux?.baz(); // undefined
```

Comment:
I don't think this should return undefined (obv that depends on the proposal itself).

I think that if you want the `new` not to throw when it gets a non-function, either the `new` or the `()` *must* have the conditional operator attached to it.

In other words, `new obj?.qux?.baz()` should be identical to `const foo = obj?.qux?.baz; new foo()`.